src/Lexer.php

```diff
@@ -825,9 +825,23 @@ protected function scanTags()
         }
 
         $token = $this->takeToken('Tag');
-        $tags = explode('@', mb_substr($line, 1, mb_strlen($line, 'utf8') - 1, 'utf8'));
-        $tags = array_map(trim(...), $tags);
-        $token['tags'] = $tags;
+
+        if ($this->compatibilityMode->shouldRemoveTagPrefixChar()) {
+            // Legacy behaviour
+            $tags = explode('@', mb_substr($line, 1, mb_strlen($line, 'utf8') - 1, 'utf8'));
+            $tags = array_map(trim(...), $tags);
+            $token['tags'] = $tags;
+
+            return $token;
+        }
+
+        $tags = preg_split('/(?=@)/u', $line);
+        assert($tags !== false);
+        // Remove the empty content before the first tag prefix
+        array_shift($tags);
+
+        // Note: checking for whitespace in tags is done in the Parser to fit with existing logic
+        $token['tags'] = array_map(trim(...), $tags);
```

Comment:
This broadly follows the logic in [cucumber/gherkin](https://github.com/cucumber/gherkin/blob/dd89f62c7ba8dc2661aa0d93de48200d07cc2df1/php/src/StringGherkinLine.php#L131-L149) except I have used a lookahead regex to save having to add the `@` back onto the split parts. It also omits checking for whitespace here as it seemed more logical to put that alongside the existing deprecation in the Parser, at least until we remove that.